src/System.Private.CoreLib/src/System/Threading/Timer.Unix.cs

```diff
@@ -13,109 +14,128 @@ namespace System.Threading
     //
     // Unix-specific implementation of Timer
     //
-    internal partial class TimerQueue
+    internal partial class TimerQueue : IThreadPoolWorkItem
     {
+        private static readonly Lock s_lock = new Lock();
```

Comment:
Would be nice to simply use standard `object s_lock = new object()` and then the `lock (s_lock)` construct. I am planning to move this code to shared partition and reuse it from Mono and possibly even CoreCLR (https://github.com/dotnet/corert/pull/6880). The performance difference is negligible now that there's indirection through `ObjectHeader` and not a hash table. I actually benchmarked it to make sure of that.